pages/api/longestStreak.ts

```diff
@@ -0,0 +1,126 @@
+import { shuffle } from "lodash";
+import { BlockState } from "../../components/math/longestStreak/MultiplicationBlock";
+import { GameLevel } from "../../redux/longestStreakSlice";
+import { getRndInteger } from "./random";
+
+export type GameBlockState = {
+  text: string;
+  value: number;
+  isProduct: boolean;
+  state: BlockState;
+};
+
+export function getRangeValues(level: GameLevel) {
+  if (level === GameLevel.BEGINNER) {
+    let a = 1
+    let b = 10
+    console.log(a, b)
+    return [a, b];
+} else if (level === GameLevel.BEGINNER_ADVANCED) {
+    let a = 10
+    let b = 20
+    return [a, b];
+  } else if (level === GameLevel.INTERMEDIATE) {
+    let a = 20
+    let b = 30
+    return [a, b];
+  } else if (level === GameLevel.INTERMEDIATE_ADVANCED) {
+    let a = 30
+    let b = 40
+    return [a, b];
+  } else if (level === GameLevel.EXPERT) {
+    let a = 40
+    let b = 50
+    return [a, b];
+  };
+}
+
+export function initializeGameState(): GameBlockState[] {
+  let dummyArray: GameBlockState[] = [];
+  for (let i = 0; i <= 19; i++) {
+      let getRangeValuesArray = getRangeValues(GameLevel.BEGINNER)
+      console.log("First range num: " + getRangeValuesArray[0])
+      console.log("second range num: " + getRangeValuesArray[1])
+      let x = getRndInteger(getRangeValuesArray[0], getRangeValuesArray[1]);
+      let y = getRndInteger(getRangeValuesArray[0], getRangeValuesArray[1]);
+      let product: number = x * y;
+      let productString: string = x + " x " + y;
+      let initiateBlockState: GameBlockState = {
+        text: product.toString(),
+        value: product,
+        isProduct: true,
+        state: BlockState.NOT_SELECTED,
+      };
+      dummyArray.push(initiateBlockState);
+
+      initiateBlockState = {
+        text: productString,
+        value: product,
+        isProduct: false,
+        state: BlockState.NOT_SELECTED,
+      };
+      dummyArray.push(initiateBlockState);
+    } 
+
+  dummyArray = shuffle(dummyArray);
+
+  return dummyArray;
+}
+
+export function longestSubarray(array: GameBlockState[], x: BlockState) {
```

Comment:
love that you moved all your logic into the api folder. your ui components are going to be so clean now. your ui is just responsible for displaying itself, no need for it to anything about the logic of the game.

moving these functions into the api folder also allows you to easily write tests for these functions, if you needed to